### PR TITLE
[PM-11903] - add file send component

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11903

## 📔 Objective

This PR adds the file send form component to the send file.

## 📸 Screenshots

|Before File Upload|After File Upload|
---|---
|Screenshot 2024-09-18 at 10 42 18 AM|Screenshot 2024-09-18 at 10 42 36 AM|

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.ts [hunk 1]
```diff
@@ -61,7 +61,8 @@ export class SendListItemsContainerComponent {
     protected toastService: ToastService,
   ) {}
 
-  async deleteSend(s: SendView): Promise<boolean> {
+  async deleteSend(e: Event, s: SendView): Promise<boolean> {
+    e.stopPropagation();
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.ts:65]: ❓ What do these changes have to do with the File Send component?

File: libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.html [hunk 2]
```diff
@@ -30,28 +30,28 @@ <h2 class="tw-font-bold" bitTypography="h6">
...
+            <button
+              bitIconButton="bwi-trash"
+              size="small"
+              type="button"
+              (click)="deleteSend($event, send)"
+              appA11yTitle="{{ 'delete' | i18n }} - {{ send.name }}"
+            ></button>
+          </bit-item-action>
+        </ng-container>
       </button>
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.html:33]: This line may be removed due to a bad merge, but I just added it with https://github.com/bitwarden/clients/pull/11120

File: apps/browser/src/tools/popup/send-v2/send-v2.component.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { RouterLink } from "@angular/router";
 import { combineLatest } from "rxjs";
 
 import { JslibModule } from "@bitwarden/angular/jslib.module";
+import { PlatformUtilsService } from "@bitwarden/common/platform/abstractions/platform-utils.service";
```
Comment (djsmith85) [inline apps/browser/src/tools/popup/send-v2/send-v2.component.ts:8]: ❓ Are these changes possibly related to other work?

File: libs/tools/send/send-ui/src/send-form/components/send-details/send-file-details.component.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    const file = (event.target as HTMLInputElement).files?.[0];
+    if (!file) {
+      return;
+    }
+    this.fileName = file.name;
+    this.sendFormContainer.onFileSelected(file);
+  };
+
+  ngOnInit() {
+    this.editMode = !!this.originalSendView;
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-form/components/send-details/send-file-details.component.ts:87]: ⚠️ The SendFormConfig provides the form mode, no need to take it form originalSendView. `config.mode === "edit"`. Can also be directly used within the markup.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.ts (hunk 1 above)
Comment (jaasen-livefront) [reply libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.ts:65]: @djsmith85 Ah yes this was a bug I discovered that was annoying me. Clicking the copy or delete buttons in the send list opens the send so this was the solution. I'll create a ticket and submit a separate PR with this.

File: apps/browser/src/tools/popup/send-v2/send-v2.component.ts (hunk 3 above)
Comment (jaasen-livefront) [reply apps/browser/src/tools/popup/send-v2/send-v2.component.ts:8]: @djsmith85 Yes you're correct! I had already implemented the requirements for https://bitwarden.atlassian.net/browse/PM-11927 while working on this feature not realizing there was a separate ticket. This was a leftover from that. I'll remove.

File: libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.html [hunk 5]
```diff
@@ -30,28 +30,27 @@ <h2 class="tw-font-bold" bitTypography="h6">
...
+            </button>
+          </bit-item-action>
+          <bit-item-action>
+            <button
+              type="button"
+              (click)="deleteSend(send)"
+              appA11yTitle="{{ 'delete' | i18n }} - {{ send.name }}"
+            >
+              <i class="bwi tw-text-lg bwi-trash"></i>
+            </button>
```
Comment (djsmith85) [inline libs/tools/send/send-ui/src/send-list-items-container/send-list-items-container.component.html:50]: ⚠️  This is removing your changes made with https://github.com/bitwarden/clients/pull/11109

**Review by djsmith85 — CHANGES_REQUESTED** [review]

**Review by djsmith85 — APPROVED** [review]
LGTM!
